torchaudio/functional.py

```diff
@@ -123,84 +140,57 @@ def spectrogram(sig, pad, window, n_fft, hop, ws, power, normalize):
     assert sig.dim() == 2
 
     if pad > 0:
-        with torch.no_grad():
-            sig = torch.nn.functional.pad(sig, (pad, pad), "constant")
-    window = window.to(sig.device)
+        # TODO add "with torch.no_grad():" back when JIT supports it
+        sig = torch.nn.functional.pad(sig, (pad, pad), "constant")
 
     # default values are consistent with librosa.core.spectrum._spectrogram
-    spec_f = torch.stft(sig, n_fft, hop, ws,
-                        window, center=True,
-                        normalized=False, onesided=True,
-                        pad_mode='reflect').transpose(1, 2)
+    spec_f = _stft(sig, n_fft, hop, ws, window,
+                   True, 'reflect', False, True).transpose(1, 2)
+
     if normalize:
         spec_f /= window.pow(2).sum().sqrt()
     spec_f = spec_f.pow(power).sum(-1)  # get power of "complex" tensor (c, l, n_fft)
     return spec_f
 
 
-def create_fb_matrix(n_stft, f_min, f_max, n_mels):
-    # type: (int, float, float, int) -> Tensor
+@torch.jit.script
+def create_fb_matrix(n_stft, sr, f_min, f_max, n_mels):
+    # type: (int, int, float, Optional[float], int) -> Tensor
     """ Create a frequency bin conversion matrix.
 
     Inputs:
         n_stft (int): number of filter banks from spectrogram
+        sr (int): sample rate of audio signal
         f_min (float): minimum frequency
-        f_max (float): maximum frequency
+        f_max (float, optional): maximum frequency. default: `sr` // 2
```

Comment:
This is marked as optional, but the argument is not a keyword argument. Is this to support "None"?

Reply:
Yes in JIT, Optional[x] means it can be type x or None. I guess for Python optional it means it has a default value e.g. ```def foo(x=1.0)```. I'll update this to be more clear

Reply:
I remember you could use None-type key argument as Option Tensor.